src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -1014,6 +1020,8 @@ bool ViewProviderSketch::mouseMove(const SbVec2s &cursorPos, Gui::View3DInventor
 {
     // maximum radius for mouse moves when selecting a geometry before switching to drag mode
     const int dragIgnoredDistance = 3;
+    prvMoveCursorPos = cursorPos;
+    prvMoveViewer = viewer;

```

Comment:
Same comments and in .h for these two.

There is a further aspect. When someone looks at this code, it appears to do nothing. This gets stored and never used. The coupling with another class via the public interface is not obvious. This also affects code readability, maintainability and increases the chances that somebody else, not understanding why this is here, may removing it or change it and break the functionality on the other end.

Reply:
I was pretty sure it wasn't correct and that you would catch it :) 
But I couldn't find a proper solution at the time.

Reply:
I need to understand first what is exactly needed. It appears to me that the solution here is a new private function in VP (possibly reusing code of the move function), with a clear name that one knows is to force a mouse move, that is exposed to DrawSketchHandler via the corresponding attorney. Then, DrawSketchHandler takes over this functionality of forcing mouse move.

The part that I understand less, is that you say that you want the tool to move the mouse in the ViewProvider. Without looking at the actual code, that sounds like code smell (if a sound can smell...odd).  The Tool dialog, is representation and input. If upon input a reaction is necessary, it should notify the specific DrawSketchHandler (either via pointer of via signal, such as a boost signal). The specific DrawSketchHandler is the one who has the responsibility to cause information to be shown in the dialog and decide what to do with the input from the dialog. The specific DrawSketchHandler should use the functionality inherited from DrawSketchHandler to trigger this in the ViewProvider via the collaboration interface.

But, I am not sure to have understood the whole picture.

Reply:
What is required is that upon input a reaction is necessary as you said. When user enter a box length, then press enter, the mousemove updates geometry by redrawing the drawEdit curve.
So when a parameter is set in the dialog, we need to be able to trigger the mouseMove function of the current DrawSketchHandler subclass.
Could you please advise how to do?
I think we could create a "onParameterSet" function, which could trigger when a spinbox has value changed. Then this function could call mouse move or do something else. 
But how to catch the signal from the dialog?